x-pack/legacy/plugins/maps/public/layers/styles/vector/components/color/color_map_select.js

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { Component, Fragment } from 'react';
+
+import { EuiSuperSelect, EuiSpacer } from '@elastic/eui';
+import { ColorStopsOrdinal } from './color_stops_ordinal';
+import { COLOR_MAP_TYPE } from '../../../../../../common/constants';
+import { ColorStopsCategorical } from './color_stops_categorical';
+
+const CUSTOM_COLOR_MAP = 'CUSTOM_COLOR_MAP';
+
+export class ColorMapSelect extends Component {
+  state = {};
+
+  static getDerivedStateFromProps(nextProps, prevState) {
+    let newState = null;
+    if (nextProps.customColorRamp !== prevState.prevPropsCustomColorRamp) {
+      newState = {
+        ...{
+          prevPropsCustomColorRamp: nextProps.customColorRamp, // reset tracker to latest value
+          customColorRamp: nextProps.customColorRamp, // reset customColorRamp to latest value
+        },
+      };
+    }
+
+    if (nextProps.customColorPalette !== prevState.prevPropsCustomColorPalette) {
+      if (!newState) {
+        newState = {};
+      }
+      newState = {
+        ...newState,
+        ...{
+          prevPropsCustomColorPalette: nextProps.customColorPalette, // reset tracker to latest value
+          customColorPalette: nextProps.customColorPalette, // reset customColorRamp to latest value
+        },
+      };
+    }
+
+    return newState;
+  }
+
+  _onColorMapSelect = selectedValue => {
+    const useCustomColorMap = selectedValue === CUSTOM_COLOR_MAP;
+    const newProps = {};
+    if (this.props.colorMapType === COLOR_MAP_TYPE.ORDINAL) {
```

Comment:
This component should not know about `useCustomColorRamp` vs `useCustomColorPalette`. Just pass in useCustomColorMap and have the onChange passed into this function handle where the state is stored in the descriptor

Reply:
How making the validation text more active:

Stop values must be unique.

OR

Values for stops must be unique.

Reply:
changes it to `Stop values must be unique`

Reply:
Typically, we use an ending period (but you should be consistent with the rest of the Maps app).  For this PR, it would be:

Stop must be a number.
Stop must be bigger than previous stop value.
Color must provide a value hex value.
Stop values must be unique.